src/status_im/ui2/screens/chat/group_details/view.cljs

```diff
@@ -5,60 +5,112 @@
+            [quo.components.safe-area :as safe-area]
+            [reagent.core :as reagent]
+            [status-im2.common.home.actions.view :as actions]
+            [status-im.ui2.screens.common.contact-list.view :as contact-list]
+            [oops.core :refer [oget]]))
 
 (defn back-button []
   [quo2/button {:type                :grey
                 :size                32
+                :width               32
                 :style               {:margin-left 20}
                 :accessibility-label :back-button
-                :on-press            #(rf/dispatch [:navigate-back])
-                :icon                true}
-   :i/arrow-left])
+                :on-press            #(rf/dispatch [:navigate-back])}
+   [quo2/icon :i/arrow-left {:color (colors/theme-colors colors/neutral-100 colors/white)}]])
 
 (defn options-button []
-  [quo2/button {:type                :grey
-                :size                32
-                :style               {:margin-right 20}
-                :accessibility-label :options-button
-                :icon                true}
-   :i/options])
+  (let [group (rf/sub [:chats/current-chat])]
+    [quo2/button {:type                :grey
+                  :size                32
+                  :width               32
+                  :style               {:margin-right 20}
+                  :accessibility-label :options-button
+                  :on-press            #(rf/dispatch [:bottom-sheet/show-sheet
+                                                      {:content (fn [] [actions/group-details-actions group])}])}
+     [quo2/icon :i/options {:color (colors/theme-colors colors/neutral-100 colors/white)}]]))
+
+(defn top-buttons []
+  [rn/view {:style {:flex-direction     :row
+                    :padding-horizontal 20
+                    :justify-content    :space-between}}
+   [back-button] [options-button]])
 
 (defn count-container [count]
   [rn/view {:style (style/count-container)}
    [quo2/text {:size   :label
                :weight :medium
                :style  {:text-align :center}} count]])
 
-(defn prepare-members [members]
-  (let [admins (filter :admin? members)
-        online (filter #(and (not (:admin? %)) (:online? %)) members)
-        offline (filter #(and (not (:admin? %)) (not (:online? %))) members)]
-    (vals (cond-> {}
-            (seq admins) (assoc :owner {:title "Owner" :data admins})
-            (seq online) (assoc :online {:title "Online" :data online})
-            (seq offline) (assoc :offline {:title "Offline" :data offline})))))
-
 (defn contacts-section-header [{:keys [title]}]
   [rn/view {:style {:padding-horizontal 20 :border-top-width 1 :border-top-color colors/neutral-20 :padding-vertical 8 :margin-top 8}}
    [quo2/text {:size   :paragraph-2
                :weight :medium
                :style  {:color (colors/theme-colors colors/neutral-50 colors/neutral-40)}} title]])
 
+(def added (reagent/atom ()))
```

Comment:
this seems like a strange way to add state. I'm not quite sure how it works in clojurescript with when this gets mounted etc but it seems a bit odd that you add an atom here and not in the render of the component using it. e.g contact-requests-sheet..

or is this approach correct?? 🤔

Reply:
@ilmotta wdyt? 🤔

Reply:
good point, it should be a subscription, or if it really super local state then it could be in let, but it never should be defined like this with `def` , yeah might some super special cases, but they are rare

Reply:
It all depends on what we want. If we want to create a reusable component, then a local state inside the function is needed, otherwise multiple instances in the same screen will compete for the same shared state and this is a recipe for disaster.

But in a namespace such as this one, having a shared state for the entire screen functions is more acceptable IMO, it's correct technically speaking.

But just like you guys, I also prefer local state inside functions in general.